### ISSUE TITLE
setRequestHeader

### ISSUE THREAD

**Issue by samcfc** [calvinmetcalf/leaflet-ajax#25]
Hi thanks for your share,
I think it will be nice to be able to .setRequestHeader from options{requestHeaders:{ key1 : value1, key2 : value2 }...}

I will test it and maybe do my first PR :).

**Comment by 9SL9** [comment]
I agree, really need this functionality to authenticate against web services that require an API key etc. Any progress on this functionality?

**Comment by calvinmetcalf** [comment]
see #33, note this is only necessary across domains, as if you are on the same domain those headers automatically get set.  Also note that it is only for ajax requests that it gets set as you can't set headers for jsonp requests

**Comment by wallmarc** [comment]
Any possibility for more progress on #33?

I need this to be able to content-negotiate on an API producing json or geojson by specifying 'accept' with value either 'application/json' or 'application/geo+json'

**Comment by calvinmetcalf** [comment]
ac197b0d5d228543fbda6d47daa26f9bc2ef2d4d

**Comment by W7PEA** [comment]
Can you confirm my syntax is proper, because I do not see the headers server-side. So I want to know I'm doing it right before i dig any deeper.

```javascript
var ajaxLayer = new L.GeoJSON.AJAX(
    layerEndpoint,{
        headers: {
            lon: currCenter.lng,
            lat: currCenter.lat,
            units: "mi",
            distance: 20,
        }
});
```
I've tried headers = {} and just the {} with my position info and I always see the data on the ajaxLayer object when I console log it, but while I do see a request on my server, I do not see the headers server side.

I have tested my server works properly by using JQuery $.getJSON() so i know if headers are present in the request I can see them in my server. 

It's worth mentioning, I tried the code from  PR #50 and it did not work either. So my next assumption is that we have a bug or i'm doing it wrong. 

Thanks!

**Comment by askpatrickw** [comment]
PR #50 does fix this.

**Comment by dave-driesmans** [comment]
This is not entirely clear to me, the docu says:
> If you want to set headers to the XMLHttpRequest set the 'headers' option equal to an object.

I thought this should work:
```
var geojsonLayer = L.geoJson.ajax("route/to/esri.json",{
    	dataType: 'json',
        headers: { 'X-Api-Key': 'jPmTlUOJHtr0JYR8yrsy....' },
});

```
but probably I'm missing something
